src/conditionals/simple_programs_with_decision_points/hard/q4/MainTest.java

```diff
@@ -0,0 +1,44 @@
+package conditionals.simple_programs_with_decision_points.hard.q4;
+
+import global.BaseTest;
+import global.exceptions.InvalidClauseException;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.PlaceHolder;
+import global.variables.clauses.StringLiteral;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+public class MainTest extends BaseTest {
+    // Parsons with distractors
+    public Clause[] testSentence() {
+        TestOption.isInputTest = true;
+        TestOption.defaultInput = "true";
+
+        return new Clause[]{
+                new StringLiteral("Is the door open or closed right now\\? \\(Enter true for open and false for closed\\)"),
+                new NewLine(),
+                new PlaceHolder()
+        };
+    }
+
+    public void runMain() {
+        DoorBoy.main(new String[0]);
+    }
+
+    static Stream<Arguments> inputProvider() {
+        return Stream.of(Arguments.of("true", "I will close the door"), Arguments.of("false", "Time to open the door"));
```

Comment:
Using input provider with only 2 cases seems like a waste, I think it'd be better with just 2 tests, one for each case

I think 2 separate tests with try-catch around the runWithInput is the better move so we can give specific feedback to the student for their mistake. (dw if this doesn't make sense, ask Abhi or I when you get to it)